alvr/client_openxr/src/stream.rs

```diff
@@ -144,27 +147,33 @@ impl StreamContext {
             None
         };
 
+        let target_resolution = if let Some(upscaling) = config.upscaling.clone() {
+            scale_resolution(config.view_resolution, upscaling)
+        } else {
+            config.view_resolution
+        };
```

Comment:
This whole if-else block should probably be moved inside compute_target_view_tesolution